### PR TITLE
Fix starting prepare timer

### PR DESCRIPTION
**Pull request by sergefdrv**
This pull request fixes an unnoticed mistake in starting the prepare timer: it was done in the wrong branch of a conditional statement. The unit test was also wrong.

### CODE REVIEW

**Review by nhoriguchi — APPROVED** [review]

**Comment by nhoriguchi** [comment]
This conflicts with changes by #122. I think that It's easy to resolve it and I can ask you of rebasing. But I feel that we can/should resolve conflict in reviewer's side, so let me share ideas about resolving conflicts.

Mainly there are 2 approaches, rebasing by proposer and resolving on `git-merge`, and they have pros/cons from some viewpoints like keeping git history clean and requirement of force-pushing to master branch. But one point I think is important is who resolves conflicts.

In "rebasing by proposer" approach (current approach,) the proposer effectively does the resolution, but sometimes it's hard because the knowledge of conflicting changes might be required for the resolution. Currently most PRs are proposed by maintainers in this project, so that's not a big issue, but "resolving on `git-merge`" approach will be important too in the future. My feeling is that the "rebasing by proposer" approach is good for large and/or complicated proposals and the "resolving on `git-merge`" approach is good for small and/or simple proposals. So we can use these two in flexible manner?

**Comment by ynamiki** [comment]
> My feeling is that the "rebasing by proposer" approach is good for large and/or complicated proposals and the "resolving on git-merge" approach is good for small and/or simple proposals. So we can choose one of these in flexible manner?

Thanks for your suggestion. I agree with it.

**Comment by sergefdrv** [comment]
Thanks for bringing up the git conflict resolution issue. I think resolving conflict by maintainers might be particularly useful for pull requests from contributors that are less familiar with the code base and for whom it may be harder to do the conflict resolution themselves. Of course, we are more than happy to support any contribution as much as we can, including helping with git conflict resolution.

On the other hand, I wonder how this conflict resolution by reviewer/maintainer would actually work. The author of PR can simply force-push an updated version in their own forked repository. I guess, if anyone else would like to do conflict resolution, they would have to open a new PR. For smaller changes, it might be too much of overhead. Do you know a better way?

**Comment by sergefdrv** [comment]
Summary of [changes](https://github.com/hyperledger-labs/minbft/compare/68383ca942f49b3940ba5c1d2e6e3b1de888b18c..a9ff5f24708c7f4c2ea5fc4a11b21080ce70d2ce): rebase with conflict resolution.

**Comment by nhoriguchi** [comment]
> On the other hand, I wonder how this conflict resolution by reviewer/maintainer would actually work. The author of PR can simply force-push an updated version in their own forked repository. I guess, if anyone else would like to do conflict resolution, they would have to open a new PR. For smaller changes, it might be too much of overhead. Do you know a better way?

I think that conflict resolution is supposed to be done by a maintainer who actually does merge operation after the PR is approved. A concrete steps of it are shown if you click "command line instructions" link in the PR page, which should be like below:

> Step 1: From your project repository, check out a new branch and test the changes.
> ~~~
> git checkout -b sergefdrv-fix-starting-prepare-timer master
> git pull https://github.com/sergefdrv/minbft.git fix-starting-prepare-timer
> ~~~
> Step 2: Merge the changes and update on GitHub.
> ~~~
> git checkout master
> git merge --no-ff sergefdrv-fix-starting-prepare-timer
> git push origin master
> ~~~

If the PR has conflicts, `git-merge` command detects them and stops, so the maintainer has to resolve them manually (typically `git-add` and `git-commit` are used). After executing `git-push` to master, GitHub will detect it and close the PR automatically. In this procedure we don't have to have another PR for conflict resolution. And I found that `git-push`ing to master branch is not "force pushing", so this operation should not be prevented by branch protection.

To make this process work well, I think that some additional guidelines will be helpful:
- when a PR with conflict is approved, the maintainer judges and feedbacks whether the conflict is resolved by `git-merge` or `git-rebase`.
- `git-push` to master branch should be used only for the case with manual conflict resolution. Otherwise use GitHub web UI.

**Comment by sergefdrv** [comment]
@Naoya-Horiguchi Thank you for elaborating on this. I think it is fine to resolve trivial merge conflicts in a merge commit. By trivial merge conflicts, I mean those where changes only interfere in context lines.

**Pull request merged**
